src/test/kotlin/com/igorwojda/list/product/challenge.kt

```diff
@@ -4,7 +4,9 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun product(list: List<Int>): Int {
-    TODO("not implemented")
+    var product = 1
```

Comment:
Challenge: reimplement this by using the following concepts:
1. Reduce operation: https://kotlinlang.org/docs/reference/collection-aggregate.html#fold-and-reduce
2. An operator: https://kotlinlang.org/docs/reference/operator-overloading.html
3. A method reference (to the above operator): https://khan.github.io/kotlin-for-python-developers/#function-references
It should be a one-liner ;)